Integrations/MailListener-POP3/MailListener-POP3.py

```diff
@@ -0,0 +1,384 @@
+
+TIME_REGEX = re.compile(r'^([\w,\d: ]*) (([+-]{1})(\d{2}):?(\d{2}))?[\s\w\(\)]*$')
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+
+def connect_pop3_server():
+    global pop3_server_conn
+
+    try:
+        if pop3_server_conn is None:
+            if SSL:
+                pop3_server_conn = poplib.POP3_SSL(SERVER, PORT)  # type: ignore
+            else:
+                pop3_server_conn = poplib.POP3(SERVER, PORT)  # type: ignore
+
+            pop3_server_conn.getwelcome()  # type: ignore
+            pop3_server_conn.user(EMAIL)  # type: ignore
+            pop3_server_conn.pass_(PASSWORD)  # type: ignore
+    except Exception as e:
+        raise e
+
+
+def close_pop3_server_connection():
+    global pop3_server_conn
+    if pop3_server_conn is not None:
+        pop3_server_conn.quit()
+        pop3_server_conn = None
+
+
+def get_user_emails():
+    connect_pop3_server()
+    (resp_message, mails_list, octets) = pop3_server_conn.list()  # type: ignore
+
+    mails = []
+    index = ''
+
+    for mail in mails_list:
+        try:
+            index = mail.split(' ')[0]
+            (resp_message, lines, octets) = pop3_server_conn.retr(index)  # type: ignore
+            msg_content = unicode(b'\r\n'.join(lines), errors='ignore').encode("utf-8")
+            msg = Parser().parsestr(msg_content)
+            msg['index'] = index
+            mails.append(msg)
+        except Exception as e:
+            raise Exception("Failed to get email with index " + index + 'from the server.\nError:' + str(e))
+
+    return mails
+
+
+def parse_time(t):
+    # there is only one time refernce is the string
+    base_time, _, sign, hours, minutes = TIME_REGEX.findall(t)[0]
+
+    if all([sign, hours, minutes]):
+        seconds = int(sign + hours) * 3600 + int(sign + minutes) * 60
+        parsed_time = datetime.strptime(
+            base_time, '%a, %d %b %Y %H:%M:%S') + timedelta(seconds=seconds)
+        return parsed_time.isoformat() + 'Z'
+    else:
+        return datetime.strptime(base_time, '%a, %d %b %Y %H:%M:%S').isoformat() + 'Z'
+
+
+def get_attachment_name(headers):
+    name = headers.get('content-description', '')
+
+    if re.match(r'^.+\..{3,5}$', name):
+        return name
+
+    content_disposition = headers.get('content-disposition', '')
+
+    if content_disposition:
+        m = re.search('filename="(.*?)"', content_disposition)
+        if m:
+            name = m.group(1)
+
+    if re.match('^.+\..{3,5}$', name):
+        return name
+
+    extension = re.match(r'.*[\\/]([\d\w]{2,4}).*', headers.get('content-type', 'txt')).group(1)  # type: ignore
+
+    return name + '.' + extension
+
+
+def parse_unicode(text):
+    arr = list(set(re.findall('((=[0-9a-fA-F]{2}){2})', text)))
+    for item in arr:
+        text = text.replace(item[0], item[0].replace('=', '').decode('hex'))
+    return text
+
+
+def parse_base64(text):
+    if re.match("^=?.*?=$", text):
+        res = re.search('=\?.*?\?[A-Z]{1}\?(.*?)\?=', text, re.IGNORECASE)
```

Comment:
no need to specify count if it's 1
```suggestion
        res = re.search('=\?.*?\?[A-Z]\?(.*?)\?=', text, re.IGNORECASE)
```